### DISCUSSION TITLE
Kornia's Gaussian Blur vs PIL's

### DISCUSSION THREAD

**Discussion by Separius** [kornia/kornia#1429] [Questions] [answered]
Hi, I want to train a MoCov2 with Korina data augmentations (I want to backprop through the images), the base model uses PIL for blurring and the sigma is a uniform variable between 0.1 and 2.0, how can I translate PIL's sigma into kernel_size and sigma of kornia?

[this question](https://stackoverflow.com/questions/62968174/for-pil-imagefilter-gaussianblur-how-what-kernel-is-used-and-does-the-radius-par) on stackoverflow is relevant.

The reason I'm asking is that by just using kornia, I'm loosing 2% (absolute) accuracy on Imagenet after 35 epochs of training (compared to pil), right now I'm using [this](https://github.com/KeremTurgutlu/self_supervised/blob/af20e89b35e687d669d58e4629101ad45b6e7407/self_supervised/augmentations.py#L13) function with s=(4,32)

**Comment by edgarriba** [comment]
the way kornia implements the gaussian kernel it's strictly following the gaussian distribution formulation
https://github.com/kornia/kornia/blob/master/kornia/filters/kernels.py#L22

In the code you point to, where is this ` s2 = int(s/4)*2+1` coming from ? /cc @ducha-aiki

**Comment by ducha-aiki** [comment, marked as answer]
@Separius PIL uses quite optimized and non-straightforward approximation of the GaussianBlur with BoxBlur. That is why, the direct conversion is hard.

However, I would advice you to try the following:
kornia_sigma = PIL sigma (PIL radius)
kornia kernel_size = make_odd(3 * kornia_sigma)

where `make_odd(x): return  x if x % 2 == 1 else x + 1`
